src/components/AdministrationPage/AdministrationPage.tsx

```diff
@@ -38,18 +44,48 @@ export const AdministrationPage = () => {
   };
 
   const [announcementMessage, setAnnouncementMessage] = useState<string>('');
+  const [etaTime, setEtaTime] = useState<string>();
+  const [isEtaNa, setIsEtaNa] = useState<boolean>(false);
+  const [announcementTouched, setAnnouncementTouched] = useState<boolean>(false);
+  const [etaTouched, setEtaTouched] = useState<boolean>(false);
   const dataContainerAnnouncement = useDataContainer(
     ({ serviceData }: IService<string>) => genericSettingsService.setAnnouncementBanner(serviceData as string),
     {
       initLoadingState: false,
     }
   );
 
+  const validateForm = () => {
+    setAnnouncementTouched(true);
+    setEtaTouched(true);
+    return !isAnnouncementInvalid() && !isEtaTimeInvalid();
+  };
+
+  const isAnnouncementInvalid = () => {
+    return isMaintenanceModeOn && announcementTouched && (!announcementMessage || announcementMessage === '');
+  };
+
+  const isEtaTimeInvalid = () => {
+    return isMaintenanceModeOn && etaTouched && !isEtaNa && (!etaTime || etaTime === '');
+  };
```

Comment:
Have you considered using useForm hook?

Reply:
useForm is over-complicated because we are not using it in any fields of the administration page. we will also need to define formConfigs for all fields, at the same time we will need to refactor all filedId / id to lowerCamelCase naming because all fields are using kebab-case for now. since the admin page is for internal use only, If we decide to use it then we will also refactor it for PNC System Version fields. we can do it in a new Jira if really necessary.

Reply:
fine